src/rules/no-missing-link-fragments.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * @fileoverview Rule to ensure link fragments (URLs that start with #) reference valid headings
+ * @author Sweta Tanwar (@SwetaTanwar)
+ */
+
+//-----------------------------------------------------------------------------
+// Imports
+//-----------------------------------------------------------------------------
+import GithubSlugger from "github-slugger";
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{
+ *   RuleOptions: [{
+ *     ignoreCase?: boolean;
+ *     allowPattern?: string;
+ *   }];
+ * }>} NoMissingLinkFragmentsRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const githubLineReferencePattern = /^L\d+(?:C\d+)?(?:-L\d+(?:C\d+)?)?$/u;
+const customHeadingIdPattern = /\{#([a-z0-9_-]+)\}\s*$/u;
```

Comment:
```suggestion
const customHeadingIdPattern = /\{#([a-z0-9_-]+)\}\s*$/u;
```

Could you update `customHeadingIdPattern` to allow a wider range of characters?

Some users may use sluggers that include non-English characters, so supporting a broader character set would be helpful.